libs/barista-components/experimental/datepicker/src/calendar-body.ts

```diff
@@ -0,0 +1,350 @@
+  @Input()
+  get minDate(): D | null {
+    return this._minDate;
+  }
+  set minDate(value: D | null) {
+    this._minDate = getValidDateOrNull(this._dateAdapter, value);
+  }
+  private _minDate: D | null = null;
+
+  /** The maximum selectable date. */
+  @Input()
+  get maxDate(): D | null {
+    return this._maxDate;
+  }
+  set maxDate(value: D | null) {
+    this._maxDate = getValidDateOrNull(this._dateAdapter, value);
+  }
+  private _maxDate: D | null = null;
+
+  /** Function used to filter whether a date is selectable or not. */
+  @Input() dateFilter: (date: D) => boolean;
+
+  @Input('aria-labelledby') ariaLabelledby: string | null;
+
+  /** Emits when a new value is selected. */
+  @Output() readonly selectedChange = new EventEmitter<D>();
+
+  /** Emits when any date is activated. */
+  @Output() readonly activeDateChange = new EventEmitter<D>();
+
+  /** @internal The names of the weekdays. */
+  _weekdays: { long: string; short: string }[];
+
+  /** @internal Grid of calendar cells representing the dates of the month. */
+  _weeks: DtCalendarCell<D>[][];
+
+  /** @internal The number of blank cells to put at the beginning for the first row. */
+  _firstRowOffset: number;
+
+  /** @internal Unique id used for the calendar body table's aria-describedby and ariaLabelledby if ariaLabelledby is not provided. */
+  _labelid = `dt-calendar-body-label-${uniqueId++}`;
+
+  /** @internal Label used for the calendar body table aria-label and description. */
+  _label = '';
+
+  constructor(
+    private _dateAdapter: DtDateAdapter<D>,
+    private _changeDetectorRef: ChangeDetectorRef,
+    private _elementRef: ElementRef<HTMLElement>,
+  ) {
+    this._activeDate = this._dateAdapter.today();
+  }
+
+  focus(): void {
+    this._elementRef.nativeElement.focus();
+  }
+
+  /**
+   * @internal
+   * Checks whether the provided date cell has the same value as the provided compare value.
+   */
+  _isSame(cell: DtCalendarCell<D>, compareValue: D): boolean {
+    return (
+      compareValue !== null &&
+      cell.rawValue !== null &&
+      this._dateAdapter.compareDate(cell.rawValue, compareValue) === 0
+    );
+  }
+
+  /**
+   * @internal
+   * Select date and emit on cell click
+   */
+  _cellClicked(cell: DtCalendarCell<D>): void {
+    this._setActiveDateAndEmit(cell.rawValue);
+    this._selectActiveDate();
+    this._changeDetectorRef.markForCheck();
+  }
+
+  /**
+   * @internal
+   * Prevent default browser behavior for calendar navigation-specific keys that could otherwise disrupt it
+   */
+  _onHostKeydown(event: KeyboardEvent): void {
+    const keyCode = _readKeyCode(event);
+    let keysToPrevent = [
+      UP_ARROW,
+      DOWN_ARROW,
+      LEFT_ARROW,
+      RIGHT_ARROW,
+      ALT,
+      PAGE_UP,
+      PAGE_DOWN,
+      ENTER,
+      SPACE,
+    ];
+
+    if (keysToPrevent.includes(keyCode)) {
+      event.preventDefault();
+    }
+  }
+
+  /**
+   * @internal
+   * Defines navigation behaviour for accessibility.
```

Comment:
The UK English version is actually 'behaviour', and 'behavior' is US, so both are fine, as long as it is consistent. I usually choose UK. So is the general convention US for our comments?

Reply:
There has been the argument in the past, that dynatrace is a US based company and it would make sense to use the US English (as much as this pains me). 
IN this case I pointed it out because you mixed it up in the same file. It does not really matter though.